### PR TITLE
Merchant Item CRUD Completes US42-48 SM && ZH

### PR DESCRIPTION
**Pull request by steve-meyers**
- Update merchant items to controller and view include activate action
- Reorganize merchant spec files to have items folder
- Add method to add a default image to Items created without images
- Add create new item form for merchant items
- Add edit item form
- Add item update functionality
- [x] All tests passing?

### CODE REVIEW

File: app/models/item.rb [hunk 1]
```diff
@@ -7,10 +8,16 @@ class Item <ApplicationRecord
   validates_presence_of :name,
                         :description,
                         :price,
-                        :image,
                         :inventory
   validates_inclusion_of :active?, :in => [true, false]
   validates_numericality_of :price, greater_than: 0
+  validates_numericality_of :inventory, greater_than: 0
+  
+  def set_defaults
```
Comment (jrsewell400) [inline app/models/item.rb:16]: I like the way default images are set here

**Review by jrsewell400 — APPROVED** [review]
Nice chunk of functionality done here, testing looks solid!  Great work guys.

**Review by jrsewell400 — APPROVED** [review]
Everything looks good to me, great work guys!

File: app/controllers/merchant/items_controller.rb [hunk 2]
```diff
@@ -1,12 +1,85 @@
...
+      update_item
+    end
+  end
+
+  def destroy
+    merchant = find_merchant_from_user
+    flash[:notice] = "#{find_item(merchant).name} was successfully deleted"
+    find_item(merchant).delete
+    redirect_to '/merchant/items'
   end
```
Comment (zachholcomb) [inline app/controllers/merchant/items_controller.rb:36]: we used a lot of helper methods to keep the controller actions clean and short

File: app/controllers/merchant/items_controller.rb [hunk 3]
```diff
@@ -1,12 +1,85 @@
...
+    @item.update(item_params)
+    if @item.save
+      redirect_to '/merchant/items'
+    else
+      flash[:notice] = @item.errors.full_messages.to_sentence
+      redirect_to "/merchant/items/#{@item.id}/edit"
+    end
+  end
+
+  def change_item_status(merchant)
```
Comment (zachholcomb) [inline app/controllers/merchant/items_controller.rb:58]: this one is to change the item's status from deactivate to activate, used query params here

File: app/models/item.rb (hunk 1 above)
Comment (zachholcomb) [reply app/models/item.rb:16]: yeah we tried to set them in the migration but rails did not really like that

File: app/views/merchant/items/edit.html.erb [hunk 4]
```diff
@@ -0,0 +1,22 @@
+<h1>Edit Item</h1>
+
```
Comment (zachholcomb) [inline app/views/merchant/items/edit.html.erb:2]: this is another good use case for partials, also would like to refactor this into form_for with resources

File: config/routes.rb [hunk 5]
```diff
@@ -63,6 +63,11 @@
   namespace :merchant do
     get '/dashboard', to: 'dashboard#show'
     get '/items', to: 'items#index'
+    patch '/items/:id', to: 'items#update'
```
Comment (zachholcomb) [inline config/routes.rb:66]: want to use resources here

File: spec/models/item_spec.rb [hunk 6]
```diff
@@ -5,8 +5,10 @@
     it { should validate_presence_of :name }
     it { should validate_presence_of :description }
     it { should validate_presence_of :price }
-    it { should validate_presence_of :image }
     it { should validate_presence_of :inventory }
+    it { should validate_numericality_of(:price).is_greater_than(0) }
```
Comment (zachholcomb) [inline spec/models/item_spec.rb:9]: now items can't be created with a price and inventory that is less than 1

**Review by elguapogordo — APPROVED** [review]

**Pull request merged**